Docs: revised deployments tutorial (#6515)
PrefectHQ/prefect: docs/tutorials/deployments.md

These all come with Prefect. You just have to configure them and set them to work. You'll see how to configure each component during this tutorial.

Optionally, you can configure [storage](/concepts/storage/) for packaging and saving your flow code and dependencies. 

## Setting up

First, create a new folder that will contain all of the files and dependencies needed by your flow deployment. This is a best practice for developing and deploying flows.

<div class="terminal">
```bash
$ mkdir prefect-tutorial
$ cd prefect-tutorial
```
</div>

You may organize your flow scripts and dependencies in any way that suits for team's needs and standards. For this tutorial, we'll keep files within this directory.

## From flow to deployment

> As noted earlier, the first ingredient of a deployment is a flow script. You've seen a few of these already, and perhaps have written a few if you've been following the tutorials. 

Reviewer: ```suggestion
As noted earlier, the first ingredient of a deployment is a flow script. You've seen a few of these already, and perhaps have written a few, if you've been following the tutorials. 
```

In the this tutorial, you'll use this flow script to create a deployment on the Prefect API. With a deployment, you can trigger ad-hoc flow runs. You could also schedule automatic flow runs that run on remote infrastructure. 

You'll create the deployment for this flow by doing the following: 

- Build a deployment YAML file that specifies settings for flow runs.
- Copy the flow script to a specified storage location from which it can be retrieved for API-triggered flow runs.
- Apply the deployment YAML settings to create a deployment on the Prefect API.
- Inspect the deployment with the Prefect CLI and Prefect UI.
- Start an ad hoc flow run based on the deployment.

## Edit the flow

In the flow definition `log_flow.py` that you created earlier, comment out or remove the last line `log_flow("Marvin")`, the call to the flow function. You don't need that anymore because Prefect will call it directly with the specified parameters when it executes your deployment.

```python hl_lines="14"
import prefect
from prefect import flow, task, get_run_logger

@task
def log_task(name):
    logger = get_run_logger()

>     logger.info(f"Hello {name}!")

Reviewer 2: ```suggestion
    logger.info("Hello %s!", name)
```

    logger.info("Prefect Version = %s 🚀", prefect.__version__)

@flow()
def log_flow(name: str):
    log_task(name)

# log_flow("Marvin")
```

## Deployment creation steps

To create a deployment from an existing flow script, there are just a few steps:

1. Use the `prefect deployment build` Prefect CLI command to build a deployment definition YAML file that settings needed to create your deployment and execute flow runs from it. This step also uploads your flow script and any supporting files to storage, if you've specified storage for the deployment.
1. Optionally, you can edit the deployment YAML file to include additional settings that are not easily specified via CLI flags. 

> 1. Use the `prefect deployment apply` Prefect CLI command to create the deployment on the API based on the settings in the deployment YAML file.

Reviewer: ```suggestion
1. Use the `prefect deployment apply` Prefect CLI command to create the deployment with the Prefect Orion server based on the settings in the deployment YAML file.
```

  definitions: null
```

> Note that the YAML configuration includes the ability to add a description, a default work queue, tags, as schedule, and more. 

Reviewer: ```suggestion
Note that the YAML configuration includes the ability to add a description, a default work queue, tags, a schedule, and more. 
```

Work queues let you organize flow runs into queues for execution. Agents pick up work from queues and execute the flows.

There is no default global agent, so to orchestrate runs of `log_flow` you need to configure one. 

In the Prefect UI, you can create a work queue by selecting the **Work Queues** page, then creating a new work queue. However, you don't need to manually create a work queue because it was created automatically when you created your deployment. If you hadn't created your deployment yet, it would be created when you start your agent. 

> Open a third terminal session, then run the `prefect agent start` command, passing a `-q test` option that tells it to pull work from the `test` work queue. Remember, we configured the deployment to use this work queue at an earlier step.

Reviewer: ```suggestion
Open an additional terminal session, then run the `prefect agent start` command, passing a `-q test` option that tells it to pull work from the `test` work queue. Remember, we configured the deployment to use this work queue at an earlier step.
```
